lumigator/python/mzai/backend/settings.py

```diff
@@ -58,7 +58,7 @@ class BackendSettings(BaseSettings):
     @property
     def S3_ENDPOINT_URL(self) -> str | None:  # noqa: N802
         # Live AWS doesn't require this but LocalStack testing does, so it's optional
-        if self.AWS_HOST is not None:
+        if self.AWS_HOST == "localstack":
```

Comment:
Why the change here?

Reply:
hi @chainlink  I left it as a draft because I planned to add more context and a readme update! Currently if you write anything in the AWS_HOST variable it will add the port
```
            return f"http://{self.AWS_HOST}:{self.S3_PORT}"
```
Because we have to actually set the AWS_HOST to `object.lga1.coreweave.com` (or whatever region/endpoint we use when we create the buckets) when using coreweave it would go into that if statement and add the port. I changed so when you set AWS_HOST to localstack, is when you actually need it and it will add the port

Reply:
I was just thinking. Probably I'll add context into the ticket (so we have it documented) but avoid doing any updates into the repo as Coreweave is very specific to our use case. With the current configuration and values in the docker-compose-local.yaml localstack works out of the box, so probably no other changes needed here

Reply:
I think we can actually simplify this a bunch.

I believe for Coreweave, we'll need to use `https` and no port (ie. `https://object.lga1.coreweave.com`

The `S3_ENDPOINT_URL` variable is used to setup the [s3 client](https://github.com/mozilla-ai/lumigator/blob/helm-unit-test/lumigator/python/mzai/backend/api/deps.py#L31)

The only use of `AWS_HOST` and `S3_PORT` is to compute this value.

I propose the following
- Change `S3_ENDPOINT_URL` to be a regular variable in settings that gets passed in directly via env var (ie. not computed)

- Update the env vars in [docker-compose-local.yaml](https://github.com/mozilla-ai/lumigator/blob/helm-unit-test/.devcontainer/docker-compose-local.yaml#L75) with the full `S3_ENDPOINT_URL` for localstack `http://localstack:4566`

- Remove all references to `AWS_HOST` and `S3_PORT`

There's some other AWS ENV Var cleanup we can do, but I'll add that to another ticket

Reply:
That sounds like a good idea! I'll make the changes and ask for approval again!

Reply:
Changes applied & more context added to the PR description, ready for review!